lib/node_modules/@stdlib/array/complex64/test/test.map.js

```diff
@@ -0,0 +1,164 @@
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var isFunction = require( '@stdlib/assert/is-function' );
+var identity = require( '@stdlib/utils/identity-function' );
+var reinterpret64 = require( '@stdlib/strided/base/reinterpret-complex64' );
+var instanceOf = require( '@stdlib/assert/instance-of' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+var Complex64 = require('@stdlib/complex/float32');
+var Float32Array = require( '@stdlib/array/float32' );
+var Complex64Array = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof Complex64Array, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `map` method', function test( t ) {
+	t.strictEqual( hasOwnProp( Complex64Array.prototype, 'map' ), true, 'has property' );
+	t.strictEqual( isFunction( Complex64Array.prototype.map ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.map.call( value, identity );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.map( value );
+		};
+	}
+});
+
+tape( 'the method returns an empty array if operating on an empty complex number array', function test( t ) {
+	var arr;
+	var out;
+
+	arr = new Complex64Array();
+	out = arr.map( identity );
+
+	t.strictEqual( out.length, 0, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method returns a new complex number array containing elements which are the result of callback function', function test( t ) {
+	var expected;
+	var actual;
+	var arr;
+
+	arr = new Complex64Array( [ 1.0, -1.0, 2.0, -2.0, 3.0, -3.0 ] );
+	expected = new Float32Array( [ 2.0, -2.0, 4.0, -4.0, 6.0, -6.0 ] );
+	actual = arr.map( scale );
+
+	t.strictEqual( instanceOf( actual, Complex64Array ), true, 'returns expected value' );
+	t.strictEqual( actual.length, expected.length/2, 'returns expected value' );
```

Comment:
```suggestion
	t.strictEqual( actual.length, expected.length/2, 'returns expected value' );
	t.notEqual( actual, arr, 'returns a new instance' );
```

The test description states that we return a "new" complex number array. We should explicitly test this assumption.